extension-support/test/aws/s3-calls-test.ts

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright 2018 Brigham Young University
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+import { expect } from 'chai';
+import * as childProcess from 'child_process';
+import 'mocha';
+import * as sinon from 'sinon';
+import awsWrapper from '../../src/aws/aws-wrapper';
+import * as s3Calls from '../../src/aws/s3-calls';
+
+describe('s3Calls', () => {
+    let sandbox: sinon.SinonSandbox;
+
+    beforeEach(() => {
+        sandbox = sinon.sandbox.create();
+    });
+
+    afterEach(() => {
+        sandbox.restore();
+    });
+
+    describe('deleteMatchingPrefix', () => {
+        it('should delete file(s) from a bucket matching a prefix', async () => {
+            const listObjectsStub = sandbox.stub(awsWrapper.s3, 'listObjectsV2').resolves({
+                Name: 'FakeBucket',
+                Contents: []
+            });
+
+            const data = await s3Calls.deleteMatchingPrefix('FakeBucket', 'FakePrefix');
+            expect(data).to.equal(null);
+            expect(listObjectsStub.callCount).to.equal(1);
+        });
+    });
+
+    describe('uploadFile', () => {
+        it('should upload the file', async () => {
+            const filePath = `${__dirname}/test-upload-file.txt`;
+
+            const uploadStub = sandbox.stub(awsWrapper.s3, 'upload').resolves({});
+
+            const uploadResponse = await s3Calls.uploadFile('handel-fake-bucket', 'my-key', filePath);
+            expect(uploadResponse).to.deep.equal({});
+            expect(uploadStub.callCount).to.equal(1);
+        });
+    });
+
+    describe('uploadDirectory', () => {
```

Comment:
Heads up - Instead of commenting out tests, you can do this:

```javascript
describe.skip('description', () => {
  it('does stuff', () => {})
})
```

[Mocha Docs](https://mochajs.org/#inclusive-tests)

Reply:
Cool, didn't know that! I forgot these were commented out from during the TypeScript migration, so I'm fixing these tests to put them back in.